evm-contracts/src/v0.7/dev/Operator.sol

```diff
@@ -213,29 +218,48 @@ contract Operator is
 
   /**
    * @notice Use this to check if a node is authorized for fulfilling requests
-   * @param node The address of the Chainlink node
+   * @param sender The address of the Chainlink node
    * @return The authorization status of the node
    */
-  function isAuthorizedSender(address node)
+  function isAuthorizedSender(address sender)
     external
     view
     override
     returns (bool)
   {
-    return s_authorizedSenders[node];
+    return s_authorizedSenders[sender];
   }
 
   /**
    * @notice Sets the fulfillment permission for a given node. Use `true` to allow, `false` to disallow.
-   * @param node The address of the Chainlink node
-   * @param allowed Bool value to determine if the node can fulfill requests
+   * @param senders The addresses of the authorized Chainlink node
    */
-  function setAuthorizedSender(address node, bool allowed)
+  function setAuthorizedSenders(address[] calldata senders)
     external
     override
     onlyOwner()
   {
-    s_authorizedSenders[node] = allowed;
+    require(senders.length > 0, "Must have at least 1 authorized sender");
+    // Set previous authorized senders to false
+    for (uint256 i = 0; i < s_authorizedSenderList.length; i++) {
```

Comment:
I am not sure if `s_authorizedSenderList.length` reads from storage every time. Might be cheaper to cache the value in a local variable. Can you write a test, possibly throwaway to check that? Can throwaway the test, just want to check.

Reply:
Great spot! 

On setting 3 authorized senders, caching the length saves 795 gas!